### DISCUSSION TITLE
Cluster Sync/Replication Status

### DISCUSSION THREAD

**Discussion by richard-scott** [emqx/emqx#10111] [General]
Is it possible to get the status of sync/replication on a node in a cluster?

For example, if I rebooted a node in the cluster, would I be able to see if the node was in sync with the rest of the cluster after it comes back online?  I want to understand how you can tell if it is safe to reboot other nodes.

Or, if I added a new node to the cluster, how can I check that this has completed the data copy from other nodes? Say for example I'm upgrading the OS Image and I want to trash the old node after bringing a new one online.

**Comment by ieQu1** [comment]
Hello,

> I want to understand how you can tell if it is safe to reboot other nodes.

Simple answer: when the EMQX node starts MQTT, dashboard and REST API listeners, it has full copy of the data. EMQX waits for the tables to be replicated locally before starting business apps.

Same applies to the situation when a new node joins the cluster: when it happens, EMQX stops business apps, copies the tables from the upstream cluster, and then restarts the business apps.

Situation becomes more complicated when optional plugins are involved. A plugin may be enabled on one node, but disabled on another, and both will be running dashboard and the listeners as usual, but only one node will have a full copy of the tables created by the plugin. 
In some cases, permanently removing the node that had the last copy of the table used by the plugin may lead to problems with starting that plugin again on another node, since EMQX remembers that the last "up to date" replica of the plugin's table resided on the node that is gone, and will wait for it come up. This is a very tricky situation to resolve, so better to avoid it by running the same plugins on all nodes in the cluster.

The above tips are only relevant for EMQX 4.* or EMQX 5.* _core_ nodes. Replicant nodes (EMQX 5.0 feature) can be started and stopped at any time without risk of data loss, however one should be careful with (re)starting too many replicant nodes simultaneously, since this will cause excessive load on the core nodes.

**Comment by richard-scott** [comment]
@ieQu1 - Thanks for the reply. So without any plugins running, if you can connect to MQTT then everything is operational :+1:

**Reply by kjellwinblad** [reply]
Yes, you are correct.
